src/components/TradeAssetSearch/hooks/useGetCustomTokensQuery.tsx

```diff
@@ -0,0 +1,61 @@
+import { type ChainId } from '@shapeshiftoss/caip'
+import { useQueries, type UseQueryResult } from '@tanstack/react-query'
+import type { TokenMetadataResponse } from 'alchemy-sdk'
+import { useCallback, useMemo } from 'react'
+import { isAddress } from 'viem'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { getAlchemyInstanceByChainId } from 'lib/alchemySdkInstance'
+
+type TokenMetadata = TokenMetadataResponse & {
+  chainId: ChainId
+  contractAddress: string
+  price: string
+}
+
+type UseGetCustomTokensQueryProps = {
+  contractAddress: string
+  chainIds: ChainId[]
+}
+
+type UseGetCustomTokenQueryReturn = {
+  data: UseQueryResult<TokenMetadata | null, Error>[]
+  isLoading: boolean
+}
+
+export const useGetCustomTokensQuery = ({
+  contractAddress,
+  chainIds,
+}: UseGetCustomTokensQueryProps): UseGetCustomTokenQueryReturn => {
+  const customTokenImportEnabled = useFeatureFlag('CustomTokenImport')
+
+  const queryKey = useMemo(() => ['customTokens', contractAddress], [contractAddress])
+
+  const getTokenMetadata = useCallback(
+    async (chainId: ChainId) => {
+      const alchemy = getAlchemyInstanceByChainId(chainId)
+      const tokenMetadataResponse = await alchemy.core.getTokenMetadata(contractAddress)
+      // TODO: get price from somewhere
+      return { ...tokenMetadataResponse, chainId, contractAddress, price: '0' }
+    },
+    [contractAddress],
+  )
+
+  const queryFn = useCallback(
+    (chainId: ChainId) =>
+      isAddress(contractAddress, { strict: false }) ? getTokenMetadata(chainId) : null,
+    [contractAddress, getTokenMetadata],
+  )
+
+  const customTokensQuery = useQueries({
+    queries: chainIds.map(chainId => ({
+      queryKey: [queryKey, chainId],
+      queryFn: () => queryFn(chainId),
+      enabled: customTokenImportEnabled,
+      staleTime: Infinity,
+    })),
+  })
+
+  const isLoading = customTokensQuery.some(query => query.isLoading)
+
+  return { data: customTokensQuery, isLoading }
```

Comment:
preferably blocking: we've got a super helpful util `mergeQueryOutputs` which can combine the react-query states from multiple queries into a single output so this hook behaves like a native react-query `useQuery`.

See `combine` in `src/pages/RFOX/hooks/useCurrentEpochRewardsQuery.ts` for an example 🍻

Reply:
Ty ser, see https://github.com/shapeshift/web/pull/7285/commits/33533853cc478b1e752fdc9f68f17ec2778d6428

Note, I had to update `mergeQueryOutputs` to change the behaviour to be more akin to `Promise.AllSettled`, as some of the queries in `customTokenQueries` are expected to fail - when they do we still want to return the queries that were successful.

Reply:
thats actually awesome